internal/data/common.go

```diff
@@ -287,30 +241,147 @@ func (am AttributeMap) Delete(key string) bool {
 	return false
 }
 
-// Insert adds the AttributeKeyValue to the map when the key does not exist.
+// Insert adds the AttributeValue to the map when the key does not exist.
+// No action is applied to the map where the key already exists.
+//
+// Important: this function should not be used if the caller has access to
+// the raw value to avoid an extra allocation.
+func (am AttributeMap) Insert(k string, v AttributeValue) {
+	if _, existing := am.Get(k); !existing {
+		*am.orig = append(*am.orig, newAttributeKeyValue(k, v))
+	}
+}
+
+// Insert adds the string Value to the map when the key does not exist.
+// No action is applied to the map where the key already exists.
+func (am AttributeMap) InsertString(k string, v string) {
```

Comment:
What is the benefit of using InsertString vs Insert?

Reply:
you don't need to do an extra allocation of the AttributeValue that will be thrown away anyway.